### PR TITLE
[CI] Enable AOT inductor FP32 accuracy test for CPU

### PR DESCRIPTION
**Pull request by WeizhuoZhang-intel**
This PR enabled AOT inductor backend FP32 accuracy check for CPU in CI workflow, which could catch AOT inductor issue at early stage.

**Test Time cost:**
| Suite       	| Precision 	| Time cost 	|
|-------------	|-----------	|-----------	|
| Huggingface 	| FP32      	|   1h12m   	|
| Timm models 	| FP32      	|   1h32m   	|
|  Torchbench 	| FP32      	|   1h40m   	|

cc @ezyang @anijain2305 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng

### CODE REVIEW

File: .github/workflows/inductor.yml [hunk 1]
```diff
@@ -174,6 +174,11 @@ jobs:
           { config: "dynamic_cpu_inductor_timm", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
           { config: "dynamic_cpu_inductor_torchbench", shard: 1, num_shards: 2, runner: "linux.12xlarge" },
           { config: "dynamic_cpu_inductor_torchbench", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
+          { config: "aot_cpu_inductor_huggingface_freezing", shard: 1, num_shards: 1, runner: "linux.12xlarge" },
```
Comment (chuanqi129) [inline .github/workflows/inductor.yml:177]: Let's change the name to `cpu_aot_inductor_xxx`

Comment (WeizhuoZhang-intel) [reply .github/workflows/inductor.yml]: Done

**Review by chuanqi129 — APPROVED** [review]

File: .ci/pytorch/test.sh [hunk 2]
```diff
@@ -543,9 +543,10 @@ test_single_dynamo_benchmark() {
     test_perf_for_dashboard "$suite" \
       "${DYNAMO_BENCHMARK_FLAGS[@]}" "$@" "${partition_flags[@]}"
   else
-    if [[ "${TEST_CONFIG}" == *aot_inductor* ]]; then
+    if [[ "${TEST_CONFIG}" == *aot_inductor* && "${TEST_CONFIG}" != *cpu_aot_inductor* ]]; then
       # Test AOTInductor with the ABI-compatible mode on CI
       # This can be removed once the ABI-compatible mode becomes default.
+      # For CPU device, we perfer non ABI-compatible mode on CI when testing AOTInductor.
```
Comment (desertfire) [inline .ci/pytorch/test.sh:549]: @jgong5 @EikanWang , can you help to prioritize the ABI-compatible mode for CPU? We expect to switch over to the ABI-compatible mode as default in H2.

**Review by desertfire — APPROVED** [review]

File: .github/workflows/inductor.yml [hunk 3]
```diff
@@ -176,6 +176,11 @@ jobs:
           { config: "dynamic_cpu_inductor_timm", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
           { config: "dynamic_cpu_inductor_torchbench", shard: 1, num_shards: 2, runner: "linux.12xlarge" },
           { config: "dynamic_cpu_inductor_torchbench", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
+          { config: "cpu_aot_inductor_huggingface_freezing", shard: 1, num_shards: 1, runner: "linux.12xlarge" },
+          { config: "cpu_aot_inductor_timm_freezing", shard: 1, num_shards: 2, runner: "linux.12xlarge" },
+          { config: "cpu_aot_inductor_timm_freezing", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
+          { config: "cpu_aot_inductor_torchbench_freezing", shard: 1, num_shards: 2, runner: "linux.12xlarge" },
+          { config: "cpu_aot_inductor_torchbench_freezing", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
```
Comment (malfet) [inline .github/workflows/inductor.yml:183]: Please add information to the PR description on how long it will take to run those tests?

**Review by malfet — APPROVED** [review]

File: .github/workflows/inductor.yml (hunk 3 above)
Comment (WeizhuoZhang-intel) [reply .github/workflows/inductor.yml:183]: Thanks for you suggestion. Time cost table has been added to the description.

**Pull request closed without merging**